File: frontend/src/types/summary.ts
export interface CategorySummary {
  category: string;
  total_amount: number;
  count: number;
}

export interface MonthlySummary {
  year: number;
  month: number;
  total_amount: number;
  total_count: number;
  categories: CategorySummary[];
}

export interface MonthOption {
  year: number;
  month: number;
  count: number;
}

File: backend/app/schemas/summary.py
from __future__ import annotations

from pydantic import BaseModel


class CategorySummary(BaseModel):
    category: str
    total_amount: float
    count: int


class MonthlySummaryResponse(BaseModel):
    year: int
    month: int
    total_amount: float
    total_count: int
    categories: list[CategorySummary]


class MonthOption(BaseModel):
    year: int
    month: int
    count: int


class MonthlyListResponse(BaseModel):
    months: list[MonthOption]
